libcxx/src/memory.cpp

```diff
@@ -132,19 +135,11 @@ __sp_mut& __get_sp_mut(const void* p) {
 
 #endif // _LIBCPP_HAS_THREADS
 
-void* align(size_t alignment, size_t size, void*& ptr, size_t& space) {
-  void* r = nullptr;
-  if (size <= space) {
-    char* p1 = static_cast<char*>(ptr);
-    char* p2 = reinterpret_cast<char*>(reinterpret_cast<uintptr_t>(p1 + (alignment - 1)) & -alignment);
-    size_t d = static_cast<size_t>(p2 - p1);
-    if (d <= space - size) {
-      r   = p2;
-      ptr = r;
-      space -= d;
-    }
-  }
-  return r;
+#if defined(_LIBCPP_EXPORT_ALIGN_SYMBOL)
+
+_LIBCPP_EXPORTED_FROM_ABI void* align(size_t alignment, size_t size, void*& ptr, size_t& space) {
+  return __align_impl(alignment, size, ptr, space);
 }
+#endif // _LIBCPP_EXPORT_ALIGN_SYMBOL
```

Comment:
Can we just move this to a file that doesn't include `<__memory/align.h>`? I'm fine with duplicating the implementation, since that function is rather simple and we won't care to touch the code in the dylib anyways. Then we don't need to do that weird `_LIBCPP_EXPORT_ALIGN_SYMBOL` dance.